File: buffer/Buffer.java
package buffer;

public class Buffer {
    public String reverseItemAtIndex(int n) {
        String[] data = new String[] { "Alice", "Bob", "Charlie", "Daisy", "Ernie", "Frank", "Gob"};
        String x = data[n];
        StringBuilder sb = new StringBuilder(x);
        return sb.reverse().toString();
    }

    public static void main(String[] args) {
        Buffer buffer = new Buffer();

        System.out.println("first item: " + buffer.reverseItemAtIndex(0));
        System.out.println("last item: " + buffer.reverseItemAtIndex(6));
        System.out.println("after data end: " + buffer.reverseItemAtIndex(7));
        System.out.println("after buffer end: " + buffer.reverseItemAtIndex(1024));
        System.out.println("before buffer start: " + buffer.reverseItemAtIndex(-1));

    }
}

File: buffer/buffer.py
def reverse_item_at_index(n):    
    data = ['Alice', 'Bob', 'Charlie', 'Daisy', 'Ernie', 'Frank', 'Gob']
    x = data[n]     
    x = x[::-1]
    return x

print("first item:", reverse_item_at_index(0))
print("last item:", reverse_item_at_index(6))
print("after data end:", reverse_item_at_index(7))
print("after buffer end:", reverse_item_at_index(1024)) 
print("before buffer start:", reverse_item_at_index(-1))
